packages/quantic/force-app/solutionExamples/main/lwc/resultSendAsEmail/resultSendAsEmail.js

```diff
@@ -0,0 +1,155 @@
+import {LightningElement, api} from 'lwc';
+import {
+  registerComponentForInit,
+  initializeWithHeadless,
+  getHeadlessBundle,
+} from 'c/quanticHeadlessLoader';
+import {buildTemplateTextFromResult} from 'c/quanticUtils';
+import sendAsEmail from '@salesforce/label/c.SendAsEmail';
+import errorWithQuickAction from '@salesforce/label/c.ErrorWithQuickAction';
+import actionIsUnavailable from '@salesforce/label/c.ActionIsUnavailable';
+import {I18nUtils} from 'c/quanticUtils';
+
+/** @typedef {import("coveo").Result} Result */
+
+export default class ResultSendAsEmail extends LightningElement {
+  actionName = 'Case.SendEmail';
+  labels = {
+    sendAsEmail,
+    errorWithQuickAction: I18nUtils.format(
+      errorWithQuickAction,
+      this.actionName
+    ),
+    actionIsUnavailable: I18nUtils.format(actionIsUnavailable, this.actionName),
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The result to send in the email body.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label = this.labels.sendAsEmail;
+  /**
+   * The template used to generate the text to insert in the body of the email.
+   * Use references to result [fields](https://docs.coveo.com/en/2036/index-content/about-fields) to get their value.
+   * @api
+   * @type {string}
+   */
+  @api textTemplate =
+    '<a href="${clickUri}">${title}</a><br/><br/><quote>${excerpt}</quote>';
+
+  iconName = 'utility:email';
+  eventName = 'insightpanel__posttofeed';
+  insertType = 'replace';
+
+  /** @type {boolean} */
+  _isLoading = false;
+  /** @type {boolean} */
+  _actionHandled = false;
+  /** @type {Function} */
+  unsubscribe;
+  /** @type {CoveoHeadlessInsight} */
+  headless;
+  /** @type {NodeJS.Timeout} */
+  timeout;
+  engine;
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+    this.addEventListener(this.eventName, this.handleSendAsEmailClick);
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener(this.eventName, this.handleSendAsEmailClick);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.engine = engine;
+
+    this.actions = {
+      ...this.headless.loadInsightAnalyticsActions(engine),
+    };
```

Comment:
Would it be better to only get the relevant action directly?

```suggestion
    const  {logCaseSendEmail} = {
      ...this.headless.loadInsightAnalyticsActions(engine),
    };
    this.actions.logCaseSendEmail = logCaseSendEmail;
```

Or something like that?